diff --git a/target/linux/ramips/dts/mt7620n_nexx_wt3020-16m.dts b/target/linux/ramips/dts/mt7620n_nexx_wt3020-16m.dts
new file mode 100644
index 0000000000..15e7734c67
--- /dev/null
+++ b/target/linux/ramips/dts/mt7620n_nexx_wt3020-16m.dts
@@ -0,0 +1,18 @@
+#include "mt7620n_nexx_wt3020.dtsi"
+
+/ {
+	compatible = "nexx,wt3020-16m", "nexx,wt3020", "ralink,mt7620n-soc";
+	model = "Nexx WT3020 (16M)";
+};
+
+&firmware {
+	reg = <0x50000 0xfb0000>;
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ohci {
+	status = "okay";
+};
diff --git a/target/linux/ramips/image/mt7620.mk b/target/linux/ramips/image/mt7620.mk
index 6089750407..c108d07834 100644
--- a/target/linux/ramips/image/mt7620.mk
+++ b/target/linux/ramips/image/mt7620.mk
@@ -875,6 +875,20 @@ define Device/nexx_wt3020-8m
 endef
 TARGET_DEVICES += nexx_wt3020-8m
 
+define Device/nexx_wt3020-16m
+  SOC := mt7620n
+  IMAGE_SIZE := 7872k
+  IMAGES += factory.bin
+  IMAGE/factory.bin := $$(sysupgrade_bin) | check-size | \
+	poray-header -B WT3020 -F 16M
+  DEVICE_VENDOR := Nexx
+  DEVICE_MODEL := WT3020
+  DEVICE_VARIANT := 16M
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci
+  SUPPORTED_DEVICES += wt3020 wt3020-16M
+endef
+TARGET_DEVICES += nexx_wt3020-16m
+
 define Device/ohyeah_oy-0001
   SOC := mt7620a
   IMAGE_SIZE := 16064k
diff --git a/target/linux/ramips/mt7620/base-files/etc/board.d/02_network b/target/linux/ramips/mt7620/base-files/etc/board.d/02_network
index f7bc41669f..898f4c749b 100644
--- a/target/linux/ramips/mt7620/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/mt7620/base-files/etc/board.d/02_network
@@ -203,6 +203,7 @@ ramips_setup_interfaces()
 			"0:lan" "1:lan" "2:lan" "3:lan" "4:wan" "6@eth0"
 		;;
 	nexx,wt3020-4m|\
+	nexx,wt3020-16m|\
 	nexx,wt3020-8m)
 		ucidef_add_switch "switch0" \
 			"4:lan" "0:wan" "6@eth0"
@@ -269,6 +270,7 @@ ramips_setup_macs()
 	netgear,jwnr2010-v5|\
 	nexx,wt3020-4m|\
 	nexx,wt3020-8m|\
+	nexx,wt3020-16m|\
 	phicomm,psg1208|\
 	planex,db-wrt01|\
 	planex,mzk-750dhp|\
diff --git a/tools/firmware-utils/patches/003-wt3020-16m.patch b/tools/firmware-utils/patches/003-wt3020-16m.patch
new file mode 100644
index 0000000000..d244737a01
--- /dev/null
+++ b/tools/firmware-utils/patches/003-wt3020-16m.patch
@@ -0,0 +1,26 @@
+diff --git a/src/mkporayfw.c b/src/mkporayfw.c
+index 43c8885..9543efe 100644
+--- a/src/mkporayfw.c
++++ b/src/mkporayfw.c
+@@ -132,6 +132,9 @@ static struct flash_layout layouts[] = {
+ 	}, {
+ 		.id		= "8M",
+ 		.fw_max_len	= 0x7c0000,
++	}, {
++		.id		= "16M",
++		.fw_max_len	= 0xfc0000,
+ 	}, {
+ 		/* terminating entry */
+ 	}
+@@ -229,6 +232,11 @@ static struct board_info boards[] = {
+                 .layout_id      = "8M",
+                 .key            = KEY_NEXX_2,
+         }, {
++                .id             = "WT3020",
++                .hw_id          = HWID_NEXX_WT3020,
++                .layout_id      = "16M",
++                .key            = KEY_NEXX_2,
++        }, {
+ 
+ 
+ 
